examples/tests/microbit_v2/main.c

```diff
@@ -0,0 +1,159 @@
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include <adc.h>
+#include <alarm.h>
+#include <ble.h>
+#include <button.h>
+#include <buzzer.h>
+#include <crc.h>
+#include <gap.h>
+#include <gpio.h>
+#include <led.h>
+#include <ninedof.h>
+#include <nrf51_serialization.h>
+#include <sound_pressure.h>
+#include <temperature.h>
+#include <timer.h>
+#include <tock.h>
+
+//          temperature works, but can't print floats
+//          buttons work -- in callback, including a call to tone_sync prevents code from resuming execution after callback
+
+// This app is modeled after the Hail test app by bradjc.
+
+static void tone_callback(void) {} // Replacing empty callback with NULL causes an error
+
+static int check_err(int retval, const char* func_name) {
+  if (retval < RETURNCODE_SUCCESS) {
+    printf("Function: %s returned error: %s\n", func_name, tock_strrcode(retval));
+    tock_exit(1);
+  }
+  return retval;
+}
+
+static void light_leds(int leds[], int len) {
+  for (int i = 0; i < len; i++) {
+    check_err(led_on(leds[i]), "led_on");
+  }
+}
+
+static void button_callback(int                            btn_num,
+                            int                            val,
+                            __attribute__ ((unused)) int   arg2,
+                            __attribute__ ((unused)) void *ud) {
+  if (!val) { // button was released, not pressed
+    return;
+  }
+  int note0 = NOTE_C4;
+  int note1 = NOTE_E4;
+  int note2 = NOTE_G4;
+  if (btn_num == 0) {
+    check_err(tone(note0 * 3, 500, tone_callback), "tone");

```

Comment:
Why the "* 3" here?

Reply:
For the `* 3`, I guess we can ignore it for now.

For the 3 buttons at once, I added a variable for tracking whether we are in a playing state or not. The problem is that queuing up more than a single note is not possible, as there is only memory for one queued note. Now notes will never be queued at all, which solves the issue.